tests/test_bootstrap.py

```diff
@@ -0,0 +1,1390 @@
+        np.testing.assert_allclose(J_dot_B_data["L32"], L32, atol=atol, rtol=rtol)
+        np.testing.assert_allclose(J_dot_B_data["alpha"], alpha, atol=atol, rtol=rtol)
+        np.testing.assert_allclose(
+            J_dot_B_data["dTeds_term"], dTeds_term, atol=atol, rtol=rtol
+        )
+        np.testing.assert_allclose(
+            J_dot_B_data["dTids_term"], dTids_term, atol=atol, rtol=rtol
+        )
+        np.testing.assert_allclose(
+            J_dot_B_data["dnds_term"], dnds_term, atol=atol, rtol=rtol
+        )
+        np.testing.assert_allclose(
+            J_dot_B_data["<J*B>"], J_dot_B_pass2, atol=atol, rtol=rtol
+        )
+
+    @pytest.mark.unit
+    def test_Redl_figures_2_3(self):
+        """Recreate plots to verify correctness.
+
+        Make sure the implementation here can roughly recover the plots
+        from figures 2 and 3 in the Redl paper.
+        """
+        for Zeff in [1, 1.8]:
+            target_nu_e_star = 4.0e-5
+            target_nu_i_star = 1.0e-5
+            # Make up some profiles
+            ne = PowerSeriesProfile([1.0e17])
+            Te = PowerSeriesProfile([1.0e5])
+            Ti_over_Te = np.sqrt(
+                4.9 * Zeff * Zeff * target_nu_e_star / (6.921 * target_nu_i_star)
+            )
+            Ti = PowerSeriesProfile([1.0e5 * Ti_over_Te])
+            rho = np.linspace(0, 1, 100) ** 2
+            rho = rho[1:]  # Avoid divide-by-0 on axis
+            helicity_N = 0
+            epsilon = rho
+            f_t = 1.46 * np.sqrt(epsilon) - 0.46 * epsilon
+            psi_edge = 68 / (2 * np.pi)
+            G = 32.0 - rho * rho  # Doesn't matter
+            R = 5.0 + 0.1 * rho * rho  # Doesn't matter
+            # Redl uses fixed values of nu_e*. To match this, I'll use
+            # a contrived iota profile that is chosen just to give the
+            # desired nu_*.
+
+            ne_rho = ne(rho)
+            Te_rho = Te(rho)
+            Zeff_rho = Zeff
+
+            # Sauter eq (18d):
+            ln_Lambda_e = 31.3 - np.log(np.sqrt(ne_rho) / Te_rho)
+
+            # Sauter eq (18b), but without the iota factor:
+            nu_e_without_iota = (
+                R
+                * (6.921e-18)
+                * ne_rho
+                * Zeff_rho
+                * ln_Lambda_e
+                / ((Te_rho**2) * (epsilon**1.5))
+            )
+
+            iota = nu_e_without_iota / target_nu_e_star
+            # End of determining the qR profile that gives the desired nu*.
+
+            geom_data = {
+                "rho": rho,
+                "G": G,
+                "R": R,
+                "iota": iota,
+                "epsilon": epsilon,
+                "f_t": f_t,
+                "psi_edge": psi_edge,
+            }
+            J_dot_B_data = j_dot_B_Redl(geom_data, ne, Te, Ti, Zeff, helicity_N)
+
+            # Change False to True in the next line to plot the data for debugging.
+            if False:
```

Comment:
could have this use `pytest-mpl` to benchmark the plot against a known correct one?

Reply:
I haven't used `pytest-mpl` before - in your experience, do the comparisons with saved reference figures work reliably across different OSs and versions of matplotlib? I'm happy to add a `pytest-mpl` check here, or alternatively `np.testing.assert_allclose` could be used to compare the computed data to reference data since that would be 100% matplotlib-version-independent. What would you prefer?

Reply:
It depends on what tolerances are used for the plot comparison, but we've found it to perform reasonably well across different versions, though we still pin the matplotlib version on CI for consistency. If it's not something that will be plotted regularly than `np.testing.assert_allclose` might make more sense, we mostly use `pytest-mpl` for the routines in `desc.plotting` to ensure that they look the way we think they should.

Reply:
`pytest-mpl` is now used here, and for the other similar spots in `test_bootstrap.py` you flagged, so there are no more `if False:` statements.